tools/parse/src/main/scala/org/allenai/nlpstack/parse/poly/decisiontree/DecisionTreeTrainer.scala

```diff
@@ -170,10 +168,23 @@ private class Node(data: Option[FeatureVectorSource], val featureVectorSubset: S
   }
 }
 
-/** Functions for training decision trees. */
-//param validationPercentage percentage of the training vectors to hold out for validation
+sealed trait InformationGainMetric {
+  val minimumGain: Double
+}
+case class EntropyGainMetric(minimumGain: Double) extends InformationGainMetric
```

Comment:
If the gain is a constant (like, always `0.0` for entropy), you can also make it a val on the class:
```
case class EntropyGainMetric extends InformationalGainMetric {
  override val minimumGain = 0.0
}
```